src/vconsole/vconsole-setup.c

```diff
@@ -234,38 +225,78 @@ static void setup_remaining_vcs(int fd, int vc_max, bool utf8) {
+        r = ioctl(fd, KDFONTOP, &cfo);
+        if (r < 0) {
+                err_ioctl = "KD_FONT_OP_GET";
+        } else {
+                unimapd.entries  = unipairs;
+                unimapd.entry_ct = USHRT_MAX;
+                r = ioctl(fd, GIO_UNIMAP, &unimapd);
+                if (r < 0)
+                        err_ioctl = "GIO_UNIMAP";
+                else
+                        cfo.op = KD_FONT_OP_SET;
+        }
+        if (r < 0)
+                log_warning_errno(errno, "%s failed, fonts will not be copied: %m", err_ioctl);
+
         for (i = 1; i <= vc_max; i++) {
-                _cleanup_close_ int vcfd = -1;
-                struct console_font_op cfo = {};
+                _cleanup_close_ int fd_d = -1;
+                char ttyname[strlen("/dev/tty") + DECIMAL_STR_MAX(int)];
 
-                if (i == vcs.v_active || !is_allocated(vcfd, i) || !is_settable(&vcfd, i))
+                if (i == vcs.v_active || !is_allocated(i))
                         continue;
 
-                (void) toggle_utf8(vcfd, utf8);
+                /* try to open terminal */
+                xsprintf(ttyname, "/dev/tty%i", i);
+                fd_d = open_terminal(ttyname, O_RDWR|O_CLOEXEC);
+                if (fd_d < 0) {
+                        log_warning_errno(errno, "Unable to open tty%i, fonts will not be copied: %m", i);
+                        continue;
+                }
 
-                /* copy font from active VT, where the font was uploaded to */
-                cfo.op = KD_FONT_OP_COPY;
-                cfo.height = vcs.v_active-1; /* tty1 == index 0 */
-                (void) ioctl(vcfd, KDFONTOP, &cfo);
+                if (!is_settable(fd_d))
+                        continue;
+
+                toggle_utf8(fd_d, utf8);
+
+                if (cfo.op != KD_FONT_OP_SET)
+                        continue;
+
+                r = ioctl(fd_d, KDFONTOP, &cfo);
+                if (r < 0) {
+                        log_warning_errno(errno, "KD_FONT_OP_SET failed, fonts will not be copied to tty%i: %m", i);
+                        continue;
+                }
 
                 /* copy unicode translation table */
                 /* unimapd is a ushort count and a pointer to an
                    array of struct unipair { ushort, ushort } */
-                unimapd.entries  = unipairs;
-                unimapd.entry_ct = USHRT_MAX;
-                if (ioctl(fd, GIO_UNIMAP, &unimapd) >= 0) {
-                        struct unimapinit adv = { 0, 0, 0 };
-
-                        (void) ioctl(vcfd, PIO_UNIMAPCLR, &adv);
-                        (void) ioctl(vcfd, PIO_UNIMAP, &unimapd);
+                r = ioctl(fd_d, PIO_UNIMAPCLR, &adv);
+                if (r < 0)
+                        err_ioctl = "PIO_UNIMAPCLR";
+                else {
+                        r = ioctl(fd_d, PIO_UNIMAP, &unimapd);
+                        if (r < 0)
+                                err_ioctl = "PIO_UNIMAP";
                 }
```

Comment:
i'd get rid of the err_ioct thing here too, and simply generate the error immediately.